src/kbucket.rs

```diff
@@ -1,219 +1,87 @@
-#![allow(unused)]
-
-use crate::helper::{Identifier, Node, U256};
+use crate::helper::{Identifier, U256};
 use std::collections::HashMap;
 use std::net::Ipv4Addr;
-use uint::*;
 
 const BUCKET_SIZE: usize = 20;
 const MAX_BUCKETS: usize = 256;
 
-type Bucket = [Option<Node>; BUCKET_SIZE];
+// Should I create a HashMap with a generic K, V or have my hashmap take in an id and TR?
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct Bucket {
+    pub map: HashMap<Identifier, TableRecord>,
+    pub limit: usize,
```

Comment:
you ignore this "limit" idea for now you can just directly use `HashMap`s so you don't have to reimplement a bunch of logic on the bucket

or you could `impl Deref` to "forward" method calls to the `map`

Reply:
https://github.com/ralexstokes/ssz-rs/blob/main/ssz-rs/src/bitlist.rs#L117